### PR TITLE
Add describeFuzz to support fuzz-specific pipeline stress runs

### PR DESCRIPTION
**Pull request by Abe27342**
## Description

This adds a simple `describe` variant that fuzz test suites can use to support environment injection of fuzz config. Currently the injected values are only how many tests to run and whether to run only the fuzz tests (and not all other tests in the package).

This describe block might be useful for devs working through fuzz test failures (allows a potentially tighter inner loop without build time), but the primary motivator is for running longer variants of our fuzz tests on CI as a part of a separate pipeline. To that end, the approach I took is pretty similar to how `describeCompat` works.

This change also converts the shared tree tests and interval collection tests to use the new `describeFuzz` infrastructure. I opted not to touch the id compressor fuzz tests since they're not currently set up to use a varying seed, but it would be easy to convert those too.

### CODE REVIEW

File: packages/test/stochastic-test-utils/src/describeFuzz.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        tests.bind(this)(testCount);
+    };
+}
+
+export type DescribeFuzzSuite = (
+    name: string,
+    tests: (
+        this: Mocha.Suite,
+        /**
+         * The number of tests this suite should produce, to within a constant factor.
```
Comment (noencke) [inline packages/test/stochastic-test-utils/src/describeFuzz.ts:20]: "to within a constant factor" - I'm not sure what this means

File: packages/test/stochastic-test-utils/src/describeFuzz.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+export const defaultOptions: Required<FuzzDescribeOptions> = {
+    defaultTestCount: 10,
+};
+
+export function createFuzzDescribe(optionsArg?: FuzzDescribeOptions): DescribeFuzz {
+    const options = { ...defaultOptions, ...optionsArg };
+    const testCountFromEnv = process.env.FUZZ_TEST_COUNT !== undefined ?
+        Number.parseInt(process.env.FUZZ_TEST_COUNT, 10) : undefined;
+    const testCount = testCountFromEnv ?? options.defaultTestCount;
+    const isStress = process.env.FUZZ_STRESS_RUN !== undefined && !!process.env.FUZZ_STRESS_RUN;
```
Comment (noencke) [inline packages/test/stochastic-test-utils/src/describeFuzz.ts:42]: You don't need the first condition here right? `!!undefined` is `false`

**Review by noencke — APPROVED** [review]

File: packages/test/stochastic-test-utils/src/describeFuzz.ts (hunk 2 above)
Comment (Abe27342) [reply packages/test/stochastic-test-utils/src/describeFuzz.ts:42]: linter complains without it, I figured might as well please the linter over disable

Comment (noencke) [reply packages/test/stochastic-test-utils/src/describeFuzz.ts:42]: ugh of course :(

**Review by noencke — APPROVED** [review]

**Pull request merged**